docs/adr/000x-modelling-history.md

```diff
@@ -0,0 +1,250 @@
+
+- comparing different editions of a document;
+- republishing live content if there are any problems (currently a common
+  support task for Whitehall publisher);
+- showing users what changes a user made in a particular edit.
+
+## Decision
+
+This ADR proposes changes to the domain model to resolve the aforementioned
+pain points and provide a means to support the future intended features. It
+defines a number of [core concepts](#core-concepts) that underpin the
+modelling. These provide means to store the individual editions of a document,
+each revision of the content of a document and each status an edition has held.
+As per [ADR-3](0003-initial-domain-modelling.md) it does not consider the
+option of sharing data between translations of a document as there are not
+the appropriate product decisions for this.
+
+A common theme in this modelling is the defining of
+[whether models are mutable](#approach-to-mutabilityimmutability) once
+persisted. Immutability is favoured as a means to preserve a history since
+each change requires storing a new record, implicitly creating a history.
+Mutability is used as a means to simplify interacting with models and preserve
+Rails idioms.
+
+This ADR then considers the impact of history for [timeline](#timeline) and
+[topics](#topics), both areas where the needs for history are less
+clear. Finally, this ADR concludes with an [overview](#overview) of the domain
+model.
+
+### Core Concepts
+
+![Main concepts](000x/main-concepts-diagram.png)
+
+**Document**: A record that represents all versions of a piece of content in a
+particular locale. It has many editions and at any time it will have a current
+edition - shown on Content Publisher index - and potentially a live edition
+which is currently on GOV.UK. The live and current edition can be
+the same. Each iteration of a document's content is represented as a revision,
+thus a document has many revisions. Document is a mutable entity that tracks
+data across all editions (such as first publishing date) and it is expected to
+be a joining point for document-related data that is not associated with a
+particular edition.
+
+**Edition**: A numbered version of a document that has been, or is
+expected to be, published on GOV.UK. It is associated with a revision
+and a status. It is mutable and is a place where any edition level
+database constraints can be placed, such as only one live edition can exist for
+a document. It is possible that two editions of the same document share the
+same revision, this is in the case that they have identical content.
+
+**Revision**: Represents an immutable snapshot of the content of a document at a
+particular point in time. It has a number to indicate which revision of the
+document it is and stores who created it. Any request by a user that changes
+content should result in a single new revision, and only data that is
+changeable by a user directly should be stored as part of a revision. This is
```

Comment:
Fair point.

I was referring to Revisions should not be used to hold data that may change independently of a users request. For example, were we to store Asset Manager state inside a revision and created new revisions each time we changed data in Asset Manager. That would be a violation of a Revision to me as it would mean that our history is confused by events that a user created and events our system created. This would prevent us from ever presenting revision history to users or revision numbers.

I've tried to rephrase this based around what we do forbid rather than what we should include.